File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque

input = sys.stdin.readline

if __name__ == "__main__":
    deque = deque()
    n = int(input())
    for _ in range(n):
        cmd = list(input().split())
        if cmd[0] == 'push_front':
            deque.appendleft(cmd[1])
        elif cmd[0] == 'push_back':
            deque.append(cmd[1])
        elif cmd[0] == 'pop_front':
            print(-1 if not deque else deque.popleft())
        elif cmd[0] == 'pop_back':
            print(-1 if not deque else deque.pop())
        elif cmd[0] == 'size':
            print(len(deque))
        elif cmd[0] == 'empty':
            print(1 if not deque else 0)
        elif cmd[0] == 'front':
            print(-1 if not deque else deque[0])
        elif cmd[0] == 'back':
            print(-1 if not deque else deque[-1])

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

public class Main {
    
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    
    
    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> deque = new LinkedList<>();
        int x;
        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            String operation = st.nextToken();
            switch (operation) {
                case "push_front":
                    x = Integer.parseInt(st.nextToken());
                    deque.addFirst(x);
                    break;
                case "push_back":
                    x = Integer.parseInt(st.nextToken());
                    deque.addLast(x);
                    break;
                case "pop_front":
                    bw.write((deque.isEmpty() ? -1 : deque.pollFirst()) + "\n");
                    break;
                case "pop_back":
                    bw.write((deque.isEmpty() ? -1 : deque.pollLast()) + "\n");
                    break;
                case "size":
                    bw.write(deque.size() + "\n");
                    break;
                case "empty":
                    bw.write((deque.isEmpty() ? 1 : 0) + "\n");
                    break;
                case "front":
                    bw.write((deque.isEmpty() ? -1 : deque.peekFirst()) + "\n");
                    break;
                case "back":
                    bw.write((deque.isEmpty() ? -1 : deque.peekLast()) + "\n");
                    break;
            }
        }
        bw.flush();
        bw.close();
    }
}